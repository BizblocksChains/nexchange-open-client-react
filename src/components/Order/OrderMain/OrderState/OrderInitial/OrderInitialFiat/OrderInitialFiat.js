import React from 'react';
import { I18n } from 'react-i18next';
import styles from '../OrderInitial.scss';

const OrderInitial = props => {
  return (
   <I18n ns="translations">
	{(t) => (
    <div id="order-payment" className={`row ${styles.container}`}>
      <div id="order-payment-details" className="col-xs-12 col-ms-6 col-sm-6 col-md-4">
        <h3>
          {t('order.initial1')}:{' '}
          <span className={styles.time}>
            <b>{props.time}</b>
          </span>
        </h3>
        <h4>
          {t('order.pay')}{' '}
          <b>
            {parseFloat(props.order.amount_quote)} {props.order.pair.quote.code}
          </b>
        </h4>

        <label>
          <input type="checkbox" name="checkboxTC" id="checkboxTC" value="check" style={{ width: '20px', height: '20px', }}
          onClick={function togglePayNowButton() {
              let _checkoutButton = document.getElementsByName("checkoutButton")[0];
              let _box = document.getElementsByName("checkboxTC")[0];
              let _box_kyc = document.getElementsByName("checkboxKYC")[0];
              let _paymentUrl = props.order.payment_url;
              if (_box.checked && _box_kyc.checked) {
                  _checkoutButton.href = _paymentUrl;
                  _checkoutButton.classList.remove("disabled");
              } else {
                  _checkoutButton.removeAttribute("href");
                  _checkoutButton.classList.add("disabled");
              }
          }}/>
          <strong style={{paddingLeft: "7px"}} dangerouslySetInnerHTML={{__html: t('order.iAgreedTC')}}/>
        </label>

        <label>
        <input type="checkbox" name="checkboxKYC" id="checkboxKYC" value="check" style={{ width: '20px', height: '20px', }}
        onClick={function togglePayNowButton() {
            let _checkoutButton = document.getElementsByName("checkoutButton")[0];
            let _box = document.getElementsByName("checkboxTC")[0];
            let _box_kyc = document.getElementsByName("checkboxKYC")[0];
            let _paymentUrl = props.order.payment_url;
            if (_box.checked && _box_kyc.checked) {
                _checkoutButton.href = _paymentUrl;
                _checkoutButton.classList.remove("disabled");
            } else {
                _checkoutButton.removeAttribute("href");
                _checkoutButton.classList.add("disabled");
            }
        }}/>
            <strong style={{paddingLeft: "7px"}}>{t('order.iAcknowledgeKYC')}</strong>
        </label>


        <a target="_blank" className="btn btn-default btn-lg disabled" name="checkoutButton" data-toggle="tooltip"
           title={t('order.tooltipTC')} style={{ pointerEvents: 'auto'}}>
          <i className="fas fa-credit-card" aria-hidden="true" style={{ position: 'relative', left: -13 }} />
          {t('order.fiat.status.pay')}
        </a>
      </div>

      <div className={`col-xs-12 col-ms-6 col-sm-6 col-md-8 ${styles.cards}`}>
        <h3>{t('order.fiat.cards')}:</h3>

        <div className="visible-xs-block visible-sm-block">
          <img src="/img/order/cards-mobile.png" alt={t('order.fiat.cardsaccepted')} />
        </div>

        <div className="visible-md-block visible-lg-block">
          <img src="/img/order/cards-desktop.png" alt={t('order.fiat.cardsaccepted')} />
        </div>
      </div>
    </div>
	)}
   </I18n>
  );
};

export default OrderInitial;
