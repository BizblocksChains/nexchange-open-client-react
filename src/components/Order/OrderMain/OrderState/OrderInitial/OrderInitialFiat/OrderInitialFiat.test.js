import React from 'react';
import { shallow } from 'enzyme';
import OrderInitialFiat from './OrderInitialFiat';
import order from 'Mocks/order';

describe('OrderInitialFiat', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<OrderInitialFiat order={order} />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
