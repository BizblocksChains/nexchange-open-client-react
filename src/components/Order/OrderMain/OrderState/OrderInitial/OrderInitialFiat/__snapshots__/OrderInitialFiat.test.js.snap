// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInitialFiat renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs-12 col-ms-6 col-sm-6 col-md-4"
          id="order-payment-details"
>
          <h3>
                    order.initial1
                    :
                     
                    <span
                              className="time"
                    >
                              <b />
                    </span>
          </h3>
          <h4>
                    order.pay
                     
                    <b>
                              0.1
                               
                              ETH
                    </b>
          </h4>
          <a
                    className="btn btn-default btn-lg"
                    href=""
                    target="_blank"
          >
                    <i
                              aria-hidden="true"
                              className="fas fa-credit-card"
                              style={
                                        Object {
                                                  "left": -13,
                                                  "position": "relative",
                                                }
                              }
                    />
                    order.fiat.status.pay
          </a>
</div>,
        <div
          className="col-xs-12 col-ms-6 col-sm-6 col-md-8 cards"
>
          <h3>
                    order.fiat.cards
                    :
          </h3>
          <div
                    className="visible-xs-block visible-sm-block"
          >
                    <img
                              alt="order.fiat.cardsaccepted"
                              src="/img/order/cards-mobile.png"
                    />
          </div>
          <div
                    className="visible-md-block visible-lg-block"
          >
                    <img
                              alt="order.fiat.cardsaccepted"
                              src="/img/order/cards-desktop.png"
                    />
          </div>
</div>,
      ],
      "className": "row container",
      "id": "order-payment",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              order.initial1
              :
               
              <span
                            className="time"
              >
                            <b />
              </span>
</h3>,
            <h4>
              order.pay
               
              <b>
                            0.1
                             
                            ETH
              </b>
</h4>,
            <a
              className="btn btn-default btn-lg"
              href=""
              target="_blank"
>
              <i
                            aria-hidden="true"
                            className="fas fa-credit-card"
                            style={
                                          Object {
                                                        "left": -13,
                                                        "position": "relative",
                                                      }
                            }
              />
              order.fiat.status.pay
</a>,
          ],
          "className": "col-xs-12 col-ms-6 col-sm-6 col-md-4",
          "id": "order-payment-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "order.initial1",
                ":",
                " ",
                <span
                  className="time"
>
                  <b />
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "order.initial1",
              ":",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b />,
                  "className": "time",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "b",
                },
                "type": "span",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "order.pay",
                " ",
                <b>
                  0.1
                   
                  ETH
</b>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "order.pay",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    0.1,
                    " ",
                    "ETH",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  0.1,
                  " ",
                  "ETH",
                ],
                "type": "b",
              },
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fas fa-credit-card"
                  style={
                                    Object {
                                                      "left": -13,
                                                      "position": "relative",
                                                    }
                  }
/>,
                "order.fiat.status.pay",
              ],
              "className": "btn btn-default btn-lg",
              "href": "",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fas fa-credit-card",
                  "style": Object {
                    "left": -13,
                    "position": "relative",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "order.fiat.status.pay",
            ],
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              order.fiat.cards
              :
</h3>,
            <div
              className="visible-xs-block visible-sm-block"
>
              <img
                            alt="order.fiat.cardsaccepted"
                            src="/img/order/cards-mobile.png"
              />
</div>,
            <div
              className="visible-md-block visible-lg-block"
>
              <img
                            alt="order.fiat.cardsaccepted"
                            src="/img/order/cards-desktop.png"
              />
</div>,
          ],
          "className": "col-xs-12 col-ms-6 col-sm-6 col-md-8 cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "order.fiat.cards",
                ":",
              ],
            },
            "ref": null,
            "rendered": Array [
              "order.fiat.cards",
              ":",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="order.fiat.cardsaccepted"
                src="/img/order/cards-mobile.png"
/>,
              "className": "visible-xs-block visible-sm-block",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "order.fiat.cardsaccepted",
                "src": "/img/order/cards-mobile.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="order.fiat.cardsaccepted"
                src="/img/order/cards-desktop.png"
/>,
              "className": "visible-md-block visible-lg-block",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "order.fiat.cardsaccepted",
                "src": "/img/order/cards-desktop.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs-12 col-ms-6 col-sm-6 col-md-4"
            id="order-payment-details"
>
            <h3>
                        order.initial1
                        :
                         
                        <span
                                    className="time"
                        >
                                    <b />
                        </span>
            </h3>
            <h4>
                        order.pay
                         
                        <b>
                                    0.1
                                     
                                    ETH
                        </b>
            </h4>
            <a
                        className="btn btn-default btn-lg"
                        href=""
                        target="_blank"
            >
                        <i
                                    aria-hidden="true"
                                    className="fas fa-credit-card"
                                    style={
                                                Object {
                                                            "left": -13,
                                                            "position": "relative",
                                                          }
                                    }
                        />
                        order.fiat.status.pay
            </a>
</div>,
          <div
            className="col-xs-12 col-ms-6 col-sm-6 col-md-8 cards"
>
            <h3>
                        order.fiat.cards
                        :
            </h3>
            <div
                        className="visible-xs-block visible-sm-block"
            >
                        <img
                                    alt="order.fiat.cardsaccepted"
                                    src="/img/order/cards-mobile.png"
                        />
            </div>
            <div
                        className="visible-md-block visible-lg-block"
            >
                        <img
                                    alt="order.fiat.cardsaccepted"
                                    src="/img/order/cards-desktop.png"
                        />
            </div>
</div>,
        ],
        "className": "row container",
        "id": "order-payment",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                order.initial1
                :
                 
                <span
                                className="time"
                >
                                <b />
                </span>
</h3>,
              <h4>
                order.pay
                 
                <b>
                                0.1
                                 
                                ETH
                </b>
</h4>,
              <a
                className="btn btn-default btn-lg"
                href=""
                target="_blank"
>
                <i
                                aria-hidden="true"
                                className="fas fa-credit-card"
                                style={
                                                Object {
                                                                "left": -13,
                                                                "position": "relative",
                                                              }
                                }
                />
                order.fiat.status.pay
</a>,
            ],
            "className": "col-xs-12 col-ms-6 col-sm-6 col-md-4",
            "id": "order-payment-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "order.initial1",
                  ":",
                  " ",
                  <span
                    className="time"
>
                    <b />
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "order.initial1",
                ":",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b />,
                    "className": "time",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "b",
                  },
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "order.pay",
                  " ",
                  <b>
                    0.1
                     
                    ETH
</b>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "order.pay",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      0.1,
                      " ",
                      "ETH",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    0.1,
                    " ",
                    "ETH",
                  ],
                  "type": "b",
                },
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fas fa-credit-card"
                    style={
                                        Object {
                                                            "left": -13,
                                                            "position": "relative",
                                                          }
                    }
/>,
                  "order.fiat.status.pay",
                ],
                "className": "btn btn-default btn-lg",
                "href": "",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fas fa-credit-card",
                    "style": Object {
                      "left": -13,
                      "position": "relative",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "order.fiat.status.pay",
              ],
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                order.fiat.cards
                :
</h3>,
              <div
                className="visible-xs-block visible-sm-block"
>
                <img
                                alt="order.fiat.cardsaccepted"
                                src="/img/order/cards-mobile.png"
                />
</div>,
              <div
                className="visible-md-block visible-lg-block"
>
                <img
                                alt="order.fiat.cardsaccepted"
                                src="/img/order/cards-desktop.png"
                />
</div>,
            ],
            "className": "col-xs-12 col-ms-6 col-sm-6 col-md-8 cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "order.fiat.cards",
                  ":",
                ],
              },
              "ref": null,
              "rendered": Array [
                "order.fiat.cards",
                ":",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="order.fiat.cardsaccepted"
                  src="/img/order/cards-mobile.png"
/>,
                "className": "visible-xs-block visible-sm-block",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "order.fiat.cardsaccepted",
                  "src": "/img/order/cards-mobile.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="order.fiat.cardsaccepted"
                  src="/img/order/cards-desktop.png"
/>,
                "className": "visible-md-block visible-lg-block",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "order.fiat.cardsaccepted",
                  "src": "/img/order/cards-desktop.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
